docs/guides/datamigrations/postgres.rst

```diff
@@ -0,0 +1,581 @@
+     public | bookmarks_id_seq | sequence | myapp
+     public | posts            | table    | myapp
+     public | posts_id_seq     | sequence | myapp
+     public | statuses         | table    | myapp
+     public | statuses_id_seq  | sequence | myapp
+     public | threads          | table    | myapp
+     public | threads_id_seq   | sequence | myapp
+     public | users            | table    | myapp
+     public | users_id_seq     | sequence | myapp
+    (11 rows)
+
+The ``users`` and ``posts`` are two tables where the bulk of the content is,
+whereas ``badges``, ``bookmarks``, ``statuses``, and ``threads`` tables store
+some meta-data that helps us tie everything together. When importing data we
+want to start with tables that are stand-alone and don't reference anything
+else and then move on to more complex data. In this case ``badges`` and
+``statuses`` are such tables, so let's look at them:
+
+.. lint-off
+
+.. code-block::
+
+    social=> \d badges
+                    Table "public.badges"
+       Column    | Type | Collation | Nullable | Default
+    -------------+------+-----------+----------+---------
+     name        | text |           | not null |
+     description | text |           | not null |
+    Indexes:
+        "badges_pkey" PRIMARY KEY, btree (name)
+    Referenced by:
+        TABLE "users" CONSTRAINT "users_badge_name_fkey" FOREIGN KEY (badge_name) REFERENCES badges(name)
+
+    social=> \d statuses
+                                Table "public.statuses"
+     Column |  Type   | Collation | Nullable |               Default
+    --------+---------+-----------+----------+--------------------------------------
+     id     | integer |           | not null | nextval('statuses_id_seq'::regclass)
+     title  | text    |           | not null |
+    Indexes:
+        "statuses_pkey" PRIMARY KEY, btree (id)
+        "statuses_title_key" UNIQUE CONSTRAINT, btree (title)
+    Referenced by:
+        TABLE "users" CONSTRAINT "users_status_id_fkey" FOREIGN KEY (status_id) REFERENCES statuses(id)
+
+.. lint-on
+
+The ``badges`` table actually uses the ``name`` of the badge as a primary key
+as opposed to a separate ``id``. So in order to reflect that properly in
+EdgeDB we would have to add :eql:constraint:`exclusive` constraint to this
+property. Meanwhile ``not null`` makes the property ``required``, so we end up
+with a type like this:
+
+.. code-block:: sql
+
+    type Badge {
+        required name: str {
+            constraint exclusive;
+        }
+        required description: str;
+    }
+
+The ``statuses`` table has a dedicated ``id`` column in addition to ``title``.
+However, the automatic ``id`` in EdgeDB is a :eql:type:`uuid`, whereas in our
+original dataset it is an ``integer``. Luckily, the ``title`` in the source
+data is ``UNIQUE``, and we might be able to use that to correctly map the data
+during our migration, leaving the old ``id`` entirely behind (assuming that no
```

Comment:
I'm curious about this part vs. the next types - is there a reason for not mapping the old id into a property like app_id (as is done later) or is it just serving as a first example of assuming that you have the id problem solved through some other means?

If the latter then I might rephrase a bit to something like "Let's assume with ``statuses`` here that we already have the old ids correctly mapped during the data, allowing us to leave the old ``id`` entirely behind..." or however you would like to put it.